docs/docsite/rst/user_guide/playbooks_variables.rst

```diff
@@ -336,11 +341,15 @@ Ansible merges different variables set in inventory so that more specific settin
 
 .. rubric:: Footnotes
 
-.. [1] Tasks in each role see their own role's defaults. Tasks defined outside of a role see the last role's defaults.
+.. [1] Tasks in each role see their own role's defaults if specified, or those from the role's dependencies. Tasks defined outside of a role see the last role's defaults.
 .. [2] Variables defined in inventory file or provided by dynamic inventory.
 .. [3] Includes vars added by 'vars plugins' as well as host_vars and group_vars which are added by the default vars plugin shipped with Ansible.
-.. [4] When created with set_facts's cacheable option, variables have the high precedence in the play,
-       but are the same as a host facts precedence when they come from the cache.
+.. [4] When created with set_facts's cacheable option, variables have the high precedence in the play, but are the same as a host facts precedence when they come from the cache.
+.. [5] Tasks in each role see their own role's vars if specified, or those from the role's dependencies. Tasks defined outside of a role see the last role's vars.
+.. [6] Tasks in each role see their own role's params if specified, or those from the role's dependencies. Role params are not exposed outside of the role.
```

Comment:
This feels a bit like mixing apples and oranges. Are `role params` really **variables**, or are they a different object, similar to playbook keywords? If they're not variables, then maybe we could describe them as `role keywords` instead of `role params`, and/or include this information in https://docs.ansible.com/ansible/latest/reference_appendices/general_precedence.html#general-precedence-rules and create better links to that page . . .

Reply:
role params are a specially scoped version of vars.

Reply:
role params (why not parameters?):
```yml
roles:
    - role: role_one
      parameter_1: 123
    - name: role_two # this is an alias for role, so never use a parameter "name" or "role"
      parameter_1: 234
```

vars:
```yaml
roles:
    - role: role_one
      vars:
        parameter_1: 123
    - name: role_two
      vars:
        name: val_of_var_name # no conflict
tasks:
  - import_role:
    role: role_3
    vars:
      var1: 345
- include_role:
  role: role_4
  vars:
    var1: 456
```
or also from `roles/x/vars/main.yml`

`parameters` and `variables` are used the same way in the role. They only have differences in the scope. (And some of this scope difference is probably unexpected, see #47606 and #36274.)